packages/protocol/contracts/L1/ProverPool.sol

```diff
@@ -380,7 +380,7 @@ contract ProverPool is EssentialContract, IProverPool {
         // by keep rewardPerGas = 1
         provers[staker.proverId] = Prover(0, 1, 0);
 
-        if (prover.stakedAmount > 0) {
+        if (prover.stakedAmount > 0 || !checkExitTimestamp) {
```

Comment:
and if a "slashed to 0 prover" called exit() by himself before his being kicked out, will his `staker.proverId` not be reset?